_posts/2020-04-17-snowplow-ios-tracker-1.3.0-released/index.md

```diff
@@ -0,0 +1,309 @@
+{% endhighlight %}
+
+<h4>1.3.2 Conditional Context Providers</h4>
+
+The previous approaches considered the generation of contexts associated to every events.
+However, there are cases where the contexts have to be applied to certain events only. 
+
+<h5>Filter Callback</h5>
+
+A filter callback is used to discriminate between events so we can attach global contexts only to certain events.
+
+{% highlight objc %}
+SPGlobalContext *filteredGC = [[SPGlobalContext alloc] initWithStaticContexts:@[[[SPSelfDescribingJson alloc] initWithSchema:@"iglu:com.snowplowanalytics.snowplow/test_sdj/jsonschema/1-0-1" andData:@{@"key": @"value"}]]
+    filter:^BOOL(id<SPInspectableEvent> event) {
+        return ["se" isEqualToString:event.name];
+    }
+];
+[tracker addGlobalContext:filteredGC tag:@"tag1"];
+{% endhighlight %}
+
+<h5>Ruleset Provider</h5>
+
+A ruleset provider is used when you want to attach a global context to certain events based on the schema URI.
+
+A ruleset provider has a ruleset which has a list of allowed schemas and a list of denied schemas. Both lists contain Iglu URIs which can be modified based on some syntactic rules.
+
+In this example, the ruleset provider will attach the generated contexts (as described in the previous section) to events with the schema `iglu:com.acme.*/*/jsonschema/*-*-*`, but not to `iglu:com.acme.marketing/*/jsonschema/*-*-*`.
+
+{% highlight objc %}
+NSString *allowed = @"iglu:com.snowplowanalytics.*/*/jsonschema/*-*-*";
+NSString *denied = @"iglu:com.snowplowanalytics.mobile/*/jsonschema/*-*-*";
+
+SPSchemaRuleset *ruleset = [SPSchemaRuleset rulesetWithAllowedList:@[allowed] andDeniedList:@[denied]];
+SPGlobalContext *rulesetGC =
+    [[SPGlobalContext alloc] initWithStaticContexts:@[[[SPSelfDescribingJson alloc] initWithSchema:@"iglu:com.snowplowanalytics.snowplow/test_sdj/jsonschema/1-0-1" andData:@{@"key": @"value"}]] ruleset:ruleset];
+[tracker addGlobalContext:rulesetGC tag:@"tag1"];
+{% endhighlight %}
+
+<h5>Ruleset format</h5>
+
+RuleSet's rules are the strings used to match against certain schemas, such as `iglu:com.acme/*/jsonschema/*-*-*`.
+
+They follow the same five-part format as an Iglu URI:
+
+{% highlight objc %}
+protocol:vendor/event_name/format/version
+{% endhighlight %}
+
+with the exception that a wildcard can be used in an allowed fashion to refer to all applying cases.
+
+The parts of a rule are wildcarded with certain guidelines:
+
+* asterisks cannot be used for the protocol (i.e. schemas always start with `iglu:`).
+
+* version matching must be specified like so: *-*-*, where any part of the versioning can be defined, e.g. 1-*-*, but only sequential parts are to be wildcarded, e.g. 1-*-1 is invalid but 1-*-* is valid.</li><li>at least two parts parts: `com.acme.*` is valid, while `com.*` is not.
+
+* vendors cannot be defined with non-wildcarded parts between wildcarded parts: com.acme.*.marketing.* is invalid, while com.acme.*.\https://zoom.us/j/779696786* is valid.
```

Comment:
Whats with the Zoom URL in the middle of this? Is that intentional?

Reply:
🤦